interface/src/app/gene/network/network.js

```diff
@@ -0,0 +1,272 @@
+      self.maxNodeNum = Number.MAX_SAFE_INTEGER;
+      self.statusMessage = 'Connecting to the server ...';
+      self.slider = {  // Slider configuration
+        position: minCorrelation,
+        options: {
+          floor: rawMinWeight,
+          ceil: rawMaxWeight,
+          step: 0.01,
+          precision: 2,
+          showSelectionBarEnd: true,
+          showTicks: 0.5,
+          showTicksValues: true,
+          onEnd: function(id, value) {
+            minCorrelation = value;
+            network.filter(minCorrelation - rawMinWeight, self.maxNodeNum)
+              .draw();
+          }
+        }
+      };
+
+      // Do nothing if no genes are specified in URL.
+      if (!$stateParams.genes || !$stateParams.genes.split(',').length) {
+        self.statusMessage = 'No genes are specified.';
+        return;
+      }
+
+      var urlGenes = $stateParams.genes.split(',');
+      var genes = [];
+      var edges = [];
+
+      /**
+       * Find index of input gene ID in genes array.
+       * @arg {int} gid Input gene's ID.
+       * @return {int} The index of gid, -1 if not found.
+       */
+      function findGeneIndex(gid) {
+        for (var i = 0; i < genes.length; ++i) {
+          if (genes[i].id === gid) {
+            return i;
+          }
+        }
+        return -1;
+      }
+
+      /**
+       * Update input edge and gene data.
+       * @param {edge} edgeIn;
+       * @param {gene} geneIn;
+       * @return {void}.
+       */
+      function updateData(edgeIn, geneIn) {
+        if (geneIn !== edgeIn.gene1 && geneIn !== edgeIn.gene2) {
+          return;
+        } // Do nothing if edgeIn and geneIn are not related.
+
+        var propertyName = (geneIn === edgeIn.gene1 ? 'source' : 'target');
+        var idx = findGeneIndex(geneIn.id);
+        if (idx === -1) {
+          if (urlGenes.indexOf(geneIn.id.toString()) !== -1) {
+            geneIn.query = true;
+          }
+          geneIn.label = (geneIn.standard_name ? geneIn.standard_name
+                          : geneIn.systematic_name);
+          genes.push(geneIn);
+          idx = genes.length - 1;
+        }
+        edgeIn[propertyName] = idx;
+        // IMPORTANT implementation detail: d3.network library will reset
+        // edge.source and edge.target to the gene object later, so the values
+        // of "source" and "target" properties of the edge won't be integers
+        // any more.
+      }
+
+      /**
+       * Draw gene-gene network.
+       * @param {void} null;
+       * @return {void}.
+       */
+      function drawNetwork() {
+        // Calculate svg size.
+        var minSvgSize = 600;  // Minimum size of svg.
+        var maxSvgSize = 1280; // Maximum size of svg.
+        var svgSize = genes.length * 10;
+        if (svgSize < minSvgSize) {
+          svgSize = minSvgSize;
+        } else if (svgSize > maxSvgSize) {
+          svgSize = maxSvgSize;
+        }
```

Comment:
The key is actually not the number of genes in the network but the number of edges. When the total number of edges is less than 300, the pre-defined maximum size seems okay. On the other hand, when a network includes so many edges, even if the size is larger, it is not easy for a user to see details. So he/she will probably have to use the minimum correlation slider bar to filter out some edges.